include/Gaffer/ContextVariables.h

```diff
@@ -55,6 +56,8 @@ class ContextVariables : public ContextProcessor<BaseType>
 		ContextVariables( const std::string &name=GraphComponent::defaultName<ContextVariables>() );
 		virtual ~ContextVariables();
 
+		StringPlug *variablesToDeletePlug();
```

Comment:
The prevailing approach throughout Gaffer has been to have one node type to create things, and another to delete those things. For example, Attributes/DeleteAttributes, PrimitiveVariables/DeletePrimitiveVariables, Outputs/DeleteOutputs. I think we should do the same here, and create a DeleteContextVariables node type rather than overload the existing node for variable creation.